ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -195,8 +197,24 @@ protected BlockCreationResult createBlock(
 
       throwIfStopped();
 
-      final Optional<List<Deposit>> maybeDeposits =
-          Optional.empty(); // TODO 6110: Extract deposits from transaction receipts
+      final DepositsValidator depositsValidator = newProtocolSpec.getDepositsValidator();
+      Optional<List<Deposit>> maybeDeposits = Optional.empty();
+
+      if (depositsValidator instanceof DepositsValidator.AllowedDeposits) {
+        final List<Deposit> depositsFromReceipts =
+            transactionResults.getReceipts().stream()
+                .flatMap(receipt -> receipt.getLogsList().stream())
+                .filter(
+                    log ->
+                        ((DepositsValidator.AllowedDeposits) depositsValidator)
+                            .getDepositContractAddress()
+                            .equals(log.getLogger()))
+                .map(DepositDecoder::decodeFromLog)
+                .toList();
+        maybeDeposits = Optional.of(depositsFromReceipts);
+      }
```

Comment:
is there a test that covers this logic?

Reply:
We definitely should have a unit test for this logic